### PR TITLE
STYLE: Extending codespell to pandas/tests

### LINKED ISSUE #38802: CI/STYLE extend codespell beyond pandas/core
There is task left here:

1. apply the following diff
```diff
diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index d78c2bacc..bcb9202d4 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -173,4 +173,4 @@ repos:
     hooks:
     -   id: codespell
         types_or: [python, rst, markdown]
         files: ^(pandas|doc)/
-        exclude: ^pandas/tests/
```
, run `pre-commit run codespell --all-files`, and fixup the mistakes 

You _may_ need you to add new words to ignore in `setup.cfg`.

### PR DESCRIPTION
**Pull request by 01-vyom**
- [X] xref #38802
- [X] tests added / passed
- [X] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/reshape/merge/test_merge_ordered.py [hunk 1]
```diff
@@ -195,7 +195,7 @@ def test_left_by_length_equals_to_right_shape0(self):
     def test_elements_not_in_by_but_in_df(self):
         # GH 38167
         left = DataFrame([["g", "h", 1], ["g", "h", 3]], columns=list("GHT"))
-        right = DataFrame([[2, 1]], columns=list("TE"))
+        right = DataFrame([[2, 1]], columns=list("T_E"))
```
Comment (01-vyom) [inline pandas/tests/reshape/merge/test_merge_ordered.py:198]: Is there any other option besides adding `TE` to ignore the list?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -16,7 +16,6 @@ repos:
     -   id: codespell
         types_or: [python, rst, markdown]
         files: ^(pandas|doc)/
-        exclude: ^pandas/tests/
```
[the remark this answers is not in the document]
Comment (01-vyom) [reply .pre-commit-config.yaml:19]: OK got it.

File: pandas/tests/io/json/test_pandas.py [hunk 3]
```diff
@@ -637,8 +637,8 @@ def test_series_non_unique_index(self):
         tm.assert_series_equal(
             s, read_json(s.to_json(orient="split"), orient="split", typ="series")
         )
-        unser = read_json(s.to_json(orient="records"), orient="records", typ="series")
-        tm.assert_numpy_array_equal(s.values, unser.values)
+        user = read_json(s.to_json(orient="records"), orient="records", typ="series")
```
Comment (MarcoGorelli) [inline pandas/tests/io/json/test_pandas.py:640]: I guess this was meant to stand for `unserialized`, perhaps rename it to that?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/tests/io/parser/common/test_common_basic.py [hunk 4]
```diff
@@ -355,7 +355,7 @@ def test_escapechar(all_parsers):
     # https://stackoverflow.com/questions/13824840/feature-request-for-
     # pandas-read-csv
     data = '''SEARCH_TERM,ACTUAL_URL
-"bra tv bord","http://www.ikea.com/se/sv/catalog/categories/departments/living_room/10475/?se%7cps%7cnonbranded%7cvardagsrum%7cgoogle%7ctv_bord"
+"bra tv board","http://www.ikea.com/se/sv/catalog/categories/departments/living_room/10475/?se%7cps%7cnonbranded%7cvardagsrum%7cgoogle%7ctv_bord"
```
Comment (attack68) [inline pandas/tests/io/parser/common/test_common_basic.py:358]: ikea is swedish and "bra tv bord" translates as "good tv stand". :)

Comment (01-vyom) [reply pandas/tests/io/parser/common/test_common_basic.py:358]: Should I ignore this word then?

Comment (MarcoGorelli) [reply pandas/tests/io/parser/common/test_common_basic.py:358]: naah, this is OK IMO, let's not ignore more than what's needed, I think this was written more in jest than anything :smile:

**Comment by 01-vyom** [comment]
I made the second PR with other files, as both of these PRs are merged I will make the final PR.

**Comment by MarcoGorelli** [comment]
Nice, thanks! There's some (unrelated to this PR) issue with CI though, will revisit once that's fixed

**Comment by MarcoGorelli** [comment]
Hi @01-vyom can you fetch and merge upstream/master please?

**Review by MarcoGorelli — APPROVED** [review]

**Comment by MarcoGorelli** [comment]
Thanks @01-vyom !

**Pull request merged**